api/v1beta1/hostedcluster_conditions.go

```diff
@@ -39,6 +39,10 @@ const (
 	// A failure here may require external user intervention to resolve. E.g. cloud provider perms were corrupted. E.g. the guest cluster was broken
 	// and kube resource deletion that affects cloud infra like service type load balancer can't succeed.
 	CloudResourcesDestroyed ConditionType = "CloudResourcesDestroyed"
+	// HostedClusterDestroyed indicates when a HostedCluster is being deleted but still needs to wait for a grace period.
+	// This grace period was introduced in order to notify when cloud dangling objects are not deleted that the user created.
+	// This condition will be grabbed by OCM personnel to report any dangling objects to the user.
```

Comment:
What's preventing the service provider / ocm from reading the `CloudResourcesDestroyed` condition, the deletion timestamp and setting / unsetting the `[CleanupCloudResourcesAnnotation](https://github.com/openshift/hypershift/blob/c00409f8236af4083b57a3f451372c8ad9ab9707/api/v1beta1/hostedcluster_types.go#L121-L124)` as they best see fit?